.github/workflows/build.yml

```diff
@@ -56,7 +56,7 @@ jobs:
       - name: Build BLE project using build.py
         run: python build.py --clean
       - name: Upload BLE hex file
-        uses: actions/upload-artifact@v1
+        uses: actions/upload-artifact@v4
         with:
-          name: build-py-BLE-${{ matrix.os }}
+          name: MICROBIT-BLE-${{ matrix.os }}-gcc-${{ matrix.gcc }}-cmake-${{ matrix.cmake || 'latest' }}
```

Comment:
`actions/upload-artifact` v4 doesn't let us overwrite uploaded artifacts, so we need to upload each one of these with a different name.